fluent/client/ws_client_test.go

```diff
@@ -0,0 +1,196 @@
+	"errors"
+
+	// "fmt"
+	// "io"
+
+	// // "os"
+	"time"
+
+	. "github.com/IBM/fluent-forward-go/fluent/client"
+	"github.com/IBM/fluent-forward-go/fluent/client/clientfakes"
+	"github.com/IBM/fluent-forward-go/fluent/client/ws/ext"
+	"github.com/IBM/fluent-forward-go/fluent/client/ws/ext/extfakes"
+	"github.com/IBM/fluent-forward-go/fluent/client/ws/wsfakes"
+	"github.com/IBM/fluent-forward-go/fluent/protocol"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("IAMAuthInfo", func() {
+	It("gets and sets an IAM token", func() {
+		iai := NewIAMAuthInfo("a")
+		Expect(iai.IAMToken()).To(Equal("a"))
+		iai.SetIAMToken("b")
+		Expect(iai.IAMToken()).To(Equal("b"))
+	})
+})
+
+var _ = Describe("WSClient", func() {
+	var (
+		factory    *clientfakes.FakeWSConnectionFactory
+		client     *WSClient
+		clientSide ext.Conn
+		conn       *wsfakes.FakeConnection
+	)
+
+	BeforeEach(func() {
+		factory = &clientfakes.FakeWSConnectionFactory{}
+		client = &WSClient{
+			ConnectionFactory: factory,
+		}
+		clientSide = &extfakes.FakeConn{}
+		conn = &wsfakes.FakeConnection{}
+
+		Expect(factory.NewCallCount()).To(Equal(0))
+		Expect(client.Session).To(BeNil())
+	})
+
+	JustBeforeEach(func() {
+		factory.NewReturns(clientSide, nil)
+	})
+
+	Describe("Connect", func() {
+		It("Does not return an error", func() {
+			Expect(client.Connect()).ToNot(HaveOccurred())
+		})
+
+		It("Gets the connection from the ConnectionFactory", func() {
+			client.Connect()
+			Expect(factory.NewCallCount()).To(Equal(1))
+		})
+
+		It("Stores the connection in the Session", func() {
+			client.Connect()
+			Expect(client.Session).ToNot(BeNil())
+			Expect(client.Session.Connection).ToNot(BeNil())
+		})
+
+		When("the factory returns an error", func() {
+			var (
+				connectionError error
+			)
+
+			JustBeforeEach(func() {
+				connectionError = errors.New("Nope")
+				factory.NewReturns(nil, connectionError)
+			})
+
+			It("Returns an error", func() {
+				err := client.Connect()
+				Expect(err).To(HaveOccurred())
+				Expect(err).To(BeIdenticalTo(connectionError))
+			})
+		})
+
+		When("the factory returns an error", func() {
+			var (
+				connectionError error
+			)
+
+			JustBeforeEach(func() {
+				connectionError = errors.New("Nope")
+				factory.NewReturns(nil, connectionError)
+			})
+
+			It("Returns an error", func() {
+				err := client.Connect()
+				Expect(err).To(HaveOccurred())
+				Expect(err).To(BeIdenticalTo(connectionError))
+			})
+		})
+	})
+
+	Describe("Disconnect", func() {
+		When("the session is not nil", func() {
+			JustBeforeEach(func() {
+				client.Connect()
+				client.Session.Connection = conn
+			})
+
+			It("closes the connection", func() {
+				Expect(client.Disconnect()).ToNot(HaveOccurred())
+				Expect(conn.CloseCallCount()).To(Equal(1))
+			})
+		})
+
+		When("the session is nil", func() {
+			JustBeforeEach(func() {
+				client.Session = nil
+			})
+
+			It("does not error or panic", func() {
+				Ω(func() {
```

Comment:
I think `Expect` is clearer/consistent over `Ω`

Reply:
`Ω` behaves differently than `Expect`. `Ω` will invoke the function that's passed to it. You can see the example of handling panics here: https://onsi.github.io/gomega/#panic

Reply:
Does it? From the source it looks to be an alias: https://github.com/onsi/gomega/blob/afd6b76d66dce7ac00ffe6eeaa351bb7b5ca7838/gomega_dsl.go#L175-L201
```
// Expect and Ω are identical
func Expect(actual interface{}, extra ...interface{}) Assertion {
	return ExpectWithOffset(0, actual, extra...)
}
```

Reply:
I see you're right. I've never seen `Expect` used with `Should`. I wonder why they aren't consistent in the docs?